### PR TITLE
Add support for `ROUTE_PREFIX` and associated tests

### PR DESCRIPTION
**Pull request by AlexMSmithCA**
#125

### CODE REVIEW

File: build/compiler.js [hunk 1]
```diff
@@ -608,8 +603,7 @@ function Compiler({
       },
       reporter: null,
       serverSideRender: true,
-      // TODO(#14): handle route prefix here??
-      publicPath: '/_static/',
+      publicPath: '/_static', // keep in sync with defaults in environment-variables-plugin.js
```
Comment (AlexMSmithCA) [inline build/compiler.js]: Only relevant change in this file.  Remainder is `prettier`'s doing.

File: plugins/route-prefix-context-plugin.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+/* eslint-env node */
+
+const envVarsPlugin = require('./environment-variables-plugin');
+
+module.exports = function() {
+  const envVars = envVarsPlugin().of();
+  return function middleware(ctx, next) {
```
Comment (AlexMSmithCA) [inline plugins/route-prefix-context-plugin.js]: Goal is to strip the prefix from `ctx.url` and supply the prefix in `ctx.prefix` for any consumers that might rely on the prefix.

**Comment by AlexMSmithCA** [comment]
I wanted to keep this backwards compatible for existing apps.  Note, however, that if a `ROUTE_PREFIX` is supplied certain plugins may not yet act as intended as they make hard-coded or implicit assumptions on route prefixes ('') and static asset ('/_static') paths.

File: build/compiler.js (hunk 1 above)
Comment (KevinGrandon) [reply build/compiler.js]: Weird, linters are failing. Do you have an older prettier version installed?

**Comment by dennisgl** [comment]
Looks good other than linting

**Pull request merged**